test/integration/controller/jobs/job/job_controller_test.go

```diff
@@ -1803,6 +1805,159 @@ var _ = ginkgo.Describe("Job controller interacting with scheduler", ginkgo.Orde
+		gomega.Expect(k8sClient.Create(ctx, equivalentFlavorB)).Should(gomega.Succeed())
+
+		equivalentClusterQ = testing.MakeClusterQueue("equivalent-cq").
+			Cohort("equivalent").
+			ResourceGroup(
+				*testing.MakeFlavorQuotas("equivalent-flavor-a").Resource(corev1.ResourceCPU, "5").Obj(),
+				*testing.MakeFlavorQuotas("equivalent-flavor-b").Resource(corev1.ResourceCPU, "5").Obj(),
+			).Obj()
+		gomega.Expect(k8sClient.Create(ctx, equivalentClusterQ)).Should(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectClusterQueueToBeDeleted(ctx, k8sClient, equivalentClusterQ, true)
+		util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, equivalentFlavorA, true)
+		util.ExpectResourceFlavorToBeDeleted(ctx, k8sClient, equivalentFlavorB, true)
+	})
+
+	ginkgo.It("Should schedule jobs as they fit in their ClusterQueue", func() {
+		ginkgo.By("creating localQueues")
+		equivalentLocalQ = testing.MakeLocalQueue("equivalent", ns.Name).ClusterQueue(equivalentClusterQ.Name).Obj()
+		gomega.Expect(k8sClient.Create(ctx, equivalentLocalQ)).Should(gomega.Succeed())
+
+		ginkgo.By("checking the first job starts")
+		// A request of 5 CPUs will fit in the first flavor.
+		equivJob1 := testingjob.MakeJob("equiv-job1", ns.Name).Queue(equivalentLocalQ.Name).Request(corev1.ResourceCPU, "5").Obj()
+		gomega.Expect(k8sClient.Create(ctx, equivJob1)).Should(gomega.Succeed())
+		lookupKey1 := types.NamespacedName{Name: equivJob1.Name, Namespace: equivJob1.Namespace}
+		createdEquivJob1 := &batchv1.Job{}
+		gomega.Eventually(func() *bool {
+			gomega.Expect(k8sClient.Get(ctx, lookupKey1, createdEquivJob1)).Should(gomega.Succeed())
+			return createdEquivJob1.Spec.Suspend
+		}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(false)))
+		gomega.Expect(createdEquivJob1.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(equivalentFlavorA.Name))
+		util.ExpectPendingWorkloadsMetric(equivalentClusterQ, 0, 0)
+
+		time.Sleep(time.Second)
+
+		ginkgo.By("checking the second job starts")
+		// A request of 5 CPUs will fit in the second flavor.
+		equivJob2 := testingjob.MakeJob("equiv-job2", ns.Name).Queue(equivalentLocalQ.Name).Request(corev1.ResourceCPU, "5").Obj()
+		gomega.Expect(k8sClient.Create(ctx, equivJob2)).Should(gomega.Succeed())
+		lookupKey2 := types.NamespacedName{Name: equivJob2.Name, Namespace: equivJob2.Namespace}
+		createdEquivJob2 := &batchv1.Job{}
+		gomega.Eventually(func() *bool {
+			gomega.Expect(k8sClient.Get(ctx, lookupKey2, createdEquivJob2)).Should(gomega.Succeed())
+			return createdEquivJob2.Spec.Suspend
+		}, util.Timeout, util.Interval).Should(gomega.Equal(ptr.To(false)))
+		gomega.Expect(createdEquivJob2.Spec.Template.Spec.NodeSelector[instanceKey]).Should(gomega.Equal(equivalentFlavorB.Name))
+		util.ExpectPendingWorkloadsMetric(equivalentClusterQ, 0, 0)
+
+		time.Sleep(time.Second)
```

Comment:
I see the first sleep is needed to make sure the creation timestamps differ. 

However, can we simplify the scenario, so that it captures the change, but uses at most one `sleep(1s)`?

Reply:
Removing these tests as per Aldo's comment